frontend/src/reducer/searchResult.js

```diff
@@ -16,16 +45,31 @@ const defaultResult = {
 }
 
 function uniqueResults (allResults) {
-  return uniqBy(allResults, 'content_id')
+  return uniqBy(allResults, 'contentId')
 }
 
 export default function searchResult (state = defaultResult, action) {
   switch (action.type) {
     case `${SET}/${SEARCH_RESULTS_LIST}`:
-      return { ...state, resultsList: uniqueResults(action.newSearchResultsList) }
+      return {
+        ...state,
+        resultsList: uniqueResults((action.newSearchResultsList).map(item => serialize(item, serializeSearchItemProps)))
+      }
+
+    case `${UPDATE}/${SEARCH_RESULTS_LIST}`:
+      return {
+        ...state,
+        resultsList: uniqueResults((state.resultsList).map(item => item.contentId === action.searchResultItem.content_id
+          ? { ...item, ...serialize(action.searchResultItem, serializeSearchItemProps) }
+          : item
+        ))
+      }
 
     case `${APPEND}/${SEARCH_RESULTS_LIST}`:
-      return { ...state, resultsList: uniqueResults([...state.resultsList, ...action.appendSearchResultsList]) }
+      return {
+        ...state,
+        resultsList: uniqueResults([...state.resultsList, ...((action.newSearchResultsList).map(item => serialize(item, serializeSearchItemProps)))])
```

Comment:
This line is starting to be a bit too complicated.
The thing that tells me that is the `)))])` at the end of it